modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/utils/getYearlyTerms.js

```diff
@@ -4,52 +4,65 @@
  */
 
 const getYearlyTerms = ({endDate, startDate}) => {
-	const yearStartDate = new Date(startDate).getFullYear();
-	const yearEndDate = new Date(endDate).getFullYear();
+	const endDateFull = new Date(endDate);
+	const startDateFull = new Date(startDate);
 
-	if (yearStartDate + 1 < yearEndDate) {
-		const yearDateSplitted = new Array(yearEndDate - yearStartDate + 1)
-			.fill()
-			.map((_, index, array) => {
-				const currentYear = yearStartDate + index;
-				const yearNumEndDate = currentYear + 1;
+	const endDateDay = endDateFull.getDate();
+	const endDateMonth = endDateFull.getMonth();
+	const endDateYear = endDateFull.getFullYear();
 
-				const yearNumStartDate = new Date(startDate).setFullYear(
-					currentYear
-				);
+	const startDateDay = startDateFull.getDate();
+	const startDateMonth = startDateFull.getMonth();
+	const startDateYear = startDateFull.getFullYear();
 
-				const daysEndDate = new Date(startDate).getDate();
-				const monthsEndDate = new Date(startDate).getMonth();
-				const hasLastElement = index + 1 === array.length;
+	if (startDateYear + 1 < endDateYear) {
+		let arraySize = endDateYear - startDateYear;
 
-				if (hasLastElement) {
-					const lastTermStartDate = new Date(yearNumStartDate);
+		if (endDateMonth > startDateMonth) {
+			arraySize = arraySize + 1;
+		}
+		else if (
+			endDateMonth === startDateMonth &&
+			endDateDay > startDateDay
+		) {
+			arraySize = arraySize + 1;
+		}
+
+		const yearDateSplitted = new Array(arraySize)
+			.fill()
+			.map((_, index, array) => {
+				const indexYear = startDateYear + index;
+				const indexNumStartDate = new Date(startDate).setFullYear(
```

Comment:
since order matters here there should be an extra line break between these to const

Reply:
that part of the code has been removed.